generated_code/viscoelastic.py

```diff
@@ -37,104 +37,75 @@
-
-# Set rules for the global matrix memory order
-stiffnessOrder = { 'Xi': 0, 'Eta': 1, 'Zeta': 2 }
-globalMatrixIdRules = [
-  (r'^k(Xi|Eta|Zeta)DivMT$', lambda x: stiffnessOrder[x[0]]),
-  (r'^k(Xi|Eta|Zeta)DivM$', lambda x: 3 + stiffnessOrder[x[0]]),  
-  (r'^fM(\d{1})$', lambda x: 6 + int(x[0])-1),
-  (r'^fP(\d{1})(\d{1})(\d{1})$', lambda x: 10 + (int(x[0])-1)*12 + (int(x[1])-1)*3 + (int(x[2])-1))
-]
-DB.determineGlobalMatrixIds(globalMatrixIdRules, db)
-
-# Kernels
-kernels = list()
-
-db.insert(DB.MatrixInfo('timeIntegrated', numberOfBasisFunctions, numberOfQuantities))
-db.insert(DB.MatrixInfo('timeDerivative0', numberOfBasisFunctions, numberOfQuantities))
-
-volume = db['kXiDivM'] * db['timeIntegrated'] * db['AstarT'] \
-       + db['kEtaDivM'] * db['timeIntegrated'] * db['BstarT'] \
-       + db['kZetaDivM'] * db['timeIntegrated'] * db['CstarT'] \
-       + db['timeIntegrated'] * db['source']
-kernels.append(('volume', volume))
-
-for i in range(0, 4):
-  localFlux = db['fM{}'.format(i+1)] * db['timeIntegrated'] * db['AplusT']
-  kernels.append(('localFlux[{}]'.format(i), localFlux))
-
-for i in range(0, 4):
-  for j in range(0, 4):
-    for h in range(0, 3):
-      neighboringFlux = db['fP{}{}{}'.format(i+1, j+1, h+1)] * db['timeIntegrated'] * db['AminusT']
-      kernels.append(('neighboringFlux[{}]'.format(i*12+j*3+h), neighboringFlux))
-
-for i in range(1, order):
-  lastD = 'timeDerivative{}'.format(str(i-1))
-  newD  = 'timeDerivative{}'.format(str(i))
-  derivative = db['kXiDivMT'] * db[lastD] * db['AstarT'] \
-             + db['kEtaDivMT'] * db[lastD] * db['BstarT'] \
-             + db['kZetaDivMT'] * db[lastD] * db['CstarT'] \
-             + db[lastD] * db['source']
-  derivative.fitBlocksToSparsityPattern()
-  kernels.append(('derivative[{}]'.format(i), derivative))
-  db.insert(derivative.flat(newD))
-  db[newD].fitBlocksToSparsityPattern()
-
-# Generate code
-Tools.generate(cmdLineArgs.outputDir, db, kernels, libxsmmGenerator, architecture)
+from yateto import Tensor
+from yateto.input import parseXMLMatrixFile, memoryLayoutFromFile
+
+from aderdg import ADERDGStandard
+
+class ADERDG(ADERDGStandard):
+  def __init__(self, order, multipleSimulations, matricesDir, memLayout, numberOfMechanisms):
+    self.numberOfMechanisms = numberOfMechanisms
+    self.numberOfElasticQuantities = 9
+
+    super().__init__(order, multipleSimulations, matricesDir)
+    clones = {
+      'star': ['star(0)', 'star(1)', 'star(2)'],
+    }
+    self.db.update( parseXMLMatrixFile('{}/matrices_viscoelastic.xml'.format(matricesDir), clones) )
+
+    star_spp = self.db.star[0].spp().as_ndarray()
+    star_rows, star_cols = star_spp.shape
+    aniso_cols = star_cols - self.numberOfElasticQuantities
+    star_spp_new = np.zeros((self.numberOfQuantities(), self.numberOfQuantities()), dtype=bool)
+    star_spp_new[0:star_rows,0:star_cols] = star_spp
+    for mech in range(1,numberOfMechanisms):
```

Comment:
Does this what you expect? range(1, 6) for example results in [1, 2, 3, 4, 5].

Reply:
Yes, basically mech 0 is repeated several times.